CsvParser/csvParser.go

```diff
@@ -129,3 +143,140 @@ func makeJsonObjFile(arr []incident.Incident) *os.File {
+	ASN := i.GetASN()
+	inc := incident.IncidentData{gpStart, gpEnd, bpStart, bpEnd, gMetric, bMetric, ASN, locationString, severity, testsAffected, gpInfo, bpInfo, iInfo}
+	return inc
+}
+
+// break the location code into approapriate locations
+func breakTheLocCodeDown(locationCode string) []string {
+	locationCodesArr := make([]string, 0)
+
+	if (len(locationCode)) > 6 {
+		for i := 0; i < 6; i = i + 2 {
+			locationCodesArr = append(locationCodesArr, locationCode[i:i+2])
+		}
+
+		locationCodesArr = append(locationCodesArr, locationCode[6:])
+
+		return locationCodesArr
+	}
+
+	for i := 0; i < len(locationCode); i = i + 2 {
+		locationCodesArr = append(locationCodesArr, locationCode[i:i+2])
+	}
+
+	return locationCodesArr
+}
+
+//check if na or eu locations directories exist for example
+func doesDirExist(originPath string, dir string) bool {
+	actualPath := originPath + "/" + dir
+
+	if _, err := os.Stat(actualPath); !os.IsNotExist(err) {
+		return true
+	}
+
+	return false
+}
+
+//check if a .json file of incidents for an asn(isp) exists in a specific location/dir
+func doesJsonFileExist(path string, asn string) (bool, string) {
+	filePath := path + "/" + asn + ".json"
+	info, err := os.Stat(filePath)
+
+	if os.IsNotExist(err) || info.IsDir() {
+		return false, filePath
+	}
+
+	return true, filePath
+}
+
+//construct location dir hierachy as you look at an incident and return the path that the incident ends up in
+func dynamicallyMakeDir(originPath string, locationCode string, asn string) string {
+	locationCodeArr := breakTheLocCodeDown(locationCode)
+	locationCodeArrLen := len(locationCodeArr)
+
+	for i := 0; i < locationCodeArrLen; i++ {
+
+		if !doesDirExist(originPath, locationCodeArr[i]) {
+			err := os.MkdirAll(originPath+"/"+locationCodeArr[i], 0755)
+
+			if err != nil {
+				log.Fatal(err)
+				//think about the implication of this error
+			}
+		}
+
+		originPath = originPath + "/" + locationCodeArr[i]
+	}
+
+	return originPath
+}
+
+func readJsonFileAddToIt(filenamepath string, i incident.DefaultIncident) {
+	jsonFile, _ := os.Open(filenamepath)
+	byteValue, _ := ioutil.ReadAll(jsonFile)
+
+	var incidents []incident.IncidentData = make([]incident.IncidentData, 0)
+
+	json.Unmarshal(byteValue, &incidents)
+
+	incidents = append(incidents, convertDefaultIncidentToIncidentData(i)) // add an incident to it
+
+	result, _ := json.Marshal(incidents)
+
+	os.Remove(filenamepath)
+
+	f, _ := os.Create(filenamepath)
+	n, err := f.Write(result)
+
+	if err != nil {
+		log.Fatal(n)
+		log.Fatal(err)
+	}
+
+	f.Close()
+
+}
+
+//uses all the helper functions above to dynamically store incidents in a file tree hierachy
+func placeIncidentInFileStruct(originPath string, i incident.DefaultIncident) {
+	//this will dynmamically create the dir to store the input incident if it needs to
+	pathToJsonFile := dynamicallyMakeDir(originPath, i.GetLocation(), i.GetASN())
+
+	fileExistance, filepath := doesJsonFileExist(pathToJsonFile, i.GetASN())
```

Comment:
this might be a bit nitpicky but you can set
location = i.GetLocation()
asn = i.GetASN() 
so the function doesn't have to be called again and again... might give MARGINALLY better performance and maybe slightly better readability

Reply:
Also, fileExistence* rather than fileExistance.